### PR TITLE
Bugfix/LTK-20031-device-is-accepting-the-fw-signature-corrupted-image…

### PR DESCRIPTION
**Pull request by PragatiGarg-eaton**
…-and-updating-the-fw:

- Firmware signature is getting added twice in the codepack/signed binary after signing the image. If we corrupt the signature at the duplicated location, FUS was successful (which makes sense too).
- Removed the redundant tlv.add function call and added UsageError message instead (referred from latest https://github.com/mcu-tools/mcuboot/blob/main/scripts/imgtool/image.py)

### CODE REVIEW

File: scripts/imgtool/image.py [hunk 1]
```diff
@@ -469,13 +469,13 @@ def create(self, public_key_format, enckey, outfile=None, certificates=None, dep
                     print("sign the digest")
                     sig = key.sign_digest(digest)
                 tlv.add(key.sig_tlv(), sig)
+                raise click.UsageError("Can not sign using key and provide fixed-signature at the same time")
```
Comment (SadanandPenore-eaton) [inline scripts/imgtool/image.py:472]: Hi @PragatiGarg-eaton , The fix was supposed to in firmware right.
Whatever wrong the code the device should not accept, right? So for the wrong image the firmware should reject it.

**Review by VineetaNarkhede-eaton — CHANGES_REQUESTED** [review]

**Review by VineetaNarkhede-eaton — APPROVED** [review]

File: scripts/imgtool/image.py (hunk 1 above)
Comment (PragatiGarg-eaton) [reply scripts/imgtool/image.py]: Image.py utility was adding firmware signatures twice in the signed binary. And, while corrupting the signature at the duplicated or wrong location, FUS was successful. So, if we corrupt the signature on the right location, where the signature resides, fus is not as successful which as expected. 
Hence, removed the redundant addition from the mcuboot which was creating the duplicate signatures.

Comment (SadanandPenore-eaton) [reply scripts/imgtool/image.py]: Call for the innersource in MCUBOOT?

**Review by SadanandPenore-eaton — APPROVED** [review]

File: scripts/imgtool/image.py [hunk 2]
```diff
@@ -474,8 +474,7 @@ def create(self, public_key_format, enckey, outfile=None, certificates=None, dep
                 tlv.add(pub_key.sig_tlv(), fixed_sig['value'])
                 self.signature = fixed_sig['value']
             else:
-                sig = key.sign_digest(digest)
-            tlv.add(key.sig_tlv(), sig)
+                raise click.UsageError("Can not sign using key and provide fixed-signature at the same time")
         else:
```
Comment (prayassamriya) [inline scripts/imgtool/image.py:478]: where are these 2 lines gone?

Comment (PragatiGarg-eaton) [reply scripts/imgtool/image.py:478]: Took a reference from the latest Zephyr 3.4 file: https://github.com/mcu-tools/mcuboot/blob/main/scripts/imgtool/image.py
However, these changes were made long back in 2022: https://github.com/mcu-tools/mcuboot/commit/160303c202ade15b8d736094a6d616f63b309153
image

File: scripts/imgtool/image.py (hunk 1 above)
Comment (PragatiGarg-eaton) [reply scripts/imgtool/image.py]: Whole cert-based code signing solution is pending to be inner-sourced. ( contributed )...

**Pull request merged**